app/controllers/newflow/login_signup_controller.rb

```diff
@@ -0,0 +1,323 @@
+          error_codes = @handler_result.errors.map(&:code)
+          security_log(:student_sign_up_failed, { reason: error_codes, email: email })
+          render :student_signup_form
+        }
+      )
+    end
+
+    def confirmation_form
+      redirect_to newflow_signup_path and return unless unverified_user.present?
+
+      @first_name = unverified_user.first_name
+      @email = unverified_user.email_addresses.first.value
+    end
+
+    def change_signup_email
+      handle_with(
+        ChangeSignupEmail,
+        user: unverified_user,
+        success: lambda {
+          redirect_to confirmation_form_updated_email_path
+        },
+        failure: lambda {
+          render :change_your_email
+        }
+      )
+    end
+
+    def confirmation_form_updated_email
+      redirect_to newflow_signup_path and return unless unverified_user.present?
+
+      @email = unverified_user.email_addresses.first.value
+    end
+
+    def verify_email_by_pin
+      handle_with(
+        VerifyEmailByPin,
+        email_address: unverified_user.email_addresses.first,
+        success: lambda {
+          clear_newflow_state
+          sign_in!(@handler_result.outputs.user)
+          security_log(:student_verified_email)
+          redirect_to signup_done_path
+        },
+        failure: lambda {
+          @first_name = unverified_user.first_name
+          @email = unverified_user.email_addresses.first.value
+          security_log(:student_verified_email_failed, email: email, user: user)
+          render :confirmation_form
+        }
+      )
+    end
+
+    def verify_email_by_code
+      handle_with(
+        VerifyEmailByCode,
+        success: lambda {
+          clear_newflow_state
+          sign_in!(@handler_result.outputs.user)
+          security_log(:student_verified_email)
+          redirect_to signup_done_path
+        },
+        failure: lambda {
+          email = @handler_result.outputs.contact_info&.value
+          user = @handler_result.outputs.user
+          security_log(:student_verified_email_failed, email: email, user: user)
+          redirect_to newflow_signup_path
+        }
+      )
+    end
+
+    def signup_done
+      @first_name = current_user.first_name
+      @email_address = current_user.email_addresses.first&.value
+    end
+
+    def profile_newflow
+      render layout: 'application'
+    end
+
+    def reset_password_form
+      @email = login_failed_email
+    end
+
+    def reset_password
+      handle_with(
+        ResetPassword,
+        success: lambda {
+          @email = @handler_result.outputs.email
+          clear_newflow_state
+          sign_out!
+          security_log :reset_password_success, user: @handler_result.outputs.user
+          render :reset_password_email_sent
+        },
+        failure: lambda {
+          security_log :reset_password_failed, user: @handler_result.outputs.user
+          render :reset_password_form
+        }
+      )
+    end
```

Comment:
I'm concerned that everything is in one file.  Maybe it is ok if everything is just `handle_with(something_common)`. But I'd like to hear more about this decision.

Reply:
I think it's better that everything is in one file. Everything _is_ for the same purpose, after all. In theory there are 4 major processes: 1) signup with password 2) login with password 3) signup with social, and 4) login with social. In practice though, the 4 are super intertwined, and with good reason. This is most evident in the old `SessionsController` and the `SessionsCreate` Lev Handler. Actually, those two things handle the overwhelming majority of the Accounts processes. 

I did try really hard to keep the one controller free of business logic, as you say `handle_with(something_common)` and what I have is the result.